src/libcore/macros.rs

```diff
@@ -229,14 +229,26 @@ macro_rules! try {
     })
 }
 
-/// Use the `format!` syntax to write data into a buffer.
+/// Calls `write_fmt` function on a writer
 ///
-/// This macro is typically used with a buffer of `&mut `[`Write`][write].
+/// This macro takes an implementor of [`std::fmt::Write`][fmt_write] or
+/// [`std::io::Write`][io_write] trait, a format string, and a list of arguments.
```

Comment:
Implementation of macro does not seem to demand writer to implement any of these traits, only to have a method called `write_fmt`.

I feel like something along the lines of

> This macro accepts any value with `write_fmt` method as a writer, a format string, and a list of arguments to format. `write_fmt` method usually comes from an implementation of [`std::fmt::Write`] or [`std::io::Write`] traits.

would be more technically correct.